core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -65,8 +65,10 @@ object ConsoleConsumer extends Logging {
   def run(conf: ConsumerConfig) {
     val timeoutMs = if (conf.timeoutMs >= 0) conf.timeoutMs else Long.MaxValue
     val consumer = new KafkaConsumer(consumerProps(conf), new ByteArrayDeserializer, new ByteArrayDeserializer)
+
+    // partition & offset is honoured only when group-id not passed.
     val consumerWrapper =
-      if (conf.partitionArg.isDefined)
+      if (!conf.groupIdPassed && conf.partitionArg.isDefined)
```

Comment:
We can remove this change now, right?

Reply:
Yes we can. But if this gets called from anywhere else then this check will help. Do you want me to remove this?

Reply:
@cmccabe Any update?

Reply:
I don't think we need this change.